substrate/threaded_queue

```diff
@@ -24,15 +24,15 @@ namespace substrate
 	public:
 		template<typename... args_t> inline void emplace(args_t &&...args) noexcept
 		{
-			std::lock_guard<std::mutex> lock{queueMutex};
+			std::unique_lock<std::mutex> lock{queueMutex};
```

Comment:
It's not that this is necessarily a wrong change, but could you explain the motivation for changing this from a lock_guard to a unique_lock please?

Reply:
I'm testing if pthread is doing some interaction between lock_guard and the condition variable that prevents the latter from properly broadcasting the wakeup signal.

Reply:
Ah, excellent